contract/src/Utils.sol

```diff
@@ -63,6 +63,38 @@ library Utils {
         return recoveredSigner == signer;
     }
 
+    /**
+     * @notice Compares two states for equality
+     * @param a The first state to compare
+     * @param b The second state to compare
+     * @return True if the states are equal, false otherwise
+     */
+    function statesAreEqual(State memory a, State memory b) internal pure returns (bool) {
```

Comment:
What's more gas efficient: compare keccaks of the states or compare field by field?